src/pages/endpoint/index.js

```diff
@@ -0,0 +1,200 @@
+import * as React from 'react'
+import styled from 'styled-components'
+import { Formik, Form } from 'formik'
+import { WithIntl } from 'components/localization'
+import Layout from 'components/layout/layout'
+import { Container, SEO } from 'components/containers'
+import { Header, Text } from 'components/elements'
+import { Input, Button } from 'components/form'
+import validation from 'common/validation'
+import { trimSpaces } from 'common/utility'
+import { default_server_url } from 'common/constants'
+import { getAppId } from 'common/websocket/config'
+import { DerivStore } from 'store'
+import { useLocalStorageState } from 'components/hooks/use-localstorage-state'
+
+const StyledContainer = styled(Container)`
+    text-align: center;
+    height: 100vh;
+    padding: auto 0;
+    justify-content: start;
+`
+
+const ButtonContainer = styled.div`
+    margin-top: 2rem;
+`
+
+const InputGroup = styled.div`
+    width: 40rem;
+    margin: 0 auto 3.4rem;
+`
+
+const StyledButton = styled(Button)`
+    margin: 0.8rem 0.4rem;
+`
+
+const endpointValidation = (values) => {
+    let errors = {}
+
+    const server_url = trimSpaces(values ? values.server_url : '')
+    const app_id = trimSpaces(values ? values.app_id.toString() : '')
+    const server_url_error = validation.text(server_url)
+    const app_id_error = validation.text(app_id)
+
+    if (server_url_error) {
+        errors.server_url = server_url_error
+    }
+
+    if (app_id_error) {
+        errors.app_id = app_id_error
+    }
+
+    return errors
+}
+
+const Endpoint = () => {
+    const [server_url, setServerUrl] = useLocalStorageState(default_server_url, 'config.server_url')
+    const [app_id, setAppId] = useLocalStorageState(getAppId(), 'config.app_id')
+    const { website_status, setWebsiteStatus, website_status_loading } = React.useContext(
+        DerivStore,
+    )
+    const TIMOUT_DELAY = 1500
+
+    const handleStatus = (setStatus, message) => {
+        setStatus({ message })
+        setTimeout(() => {
+            setStatus({})
+        }, TIMOUT_DELAY)
+    }
+    const resetEndpointSettings = (setStatus) => {
+        setServerUrl(default_server_url)
+        setAppId(getAppId())
+        setWebsiteStatus()
+        handleStatus(setStatus, 'Config has been reset successfully')
+        // TODO: if there is a change requires reload in the future
+        // window.location.reload()
+    }
+    const endpointSubmission = (values, actions) => {
+        actions.setSubmitting(true)
+        setServerUrl(values.server_url)
+        setAppId(values.app_id)
+
+        // handle website status changes
+        const new_website_status = { ...website_status, clients_country: values.clients_country }
+        setWebsiteStatus(new_website_status)
+        actions.setSubmitting(false)
+        handleStatus(actions.setStatus, 'Config has been updated')
```

Comment:
Should this text be localized?

Reply:
this page is only meant for us, no translation needed here